### DISCUSSION TITLE
multiple actions needed on one node, onElementClick alone can't distinguish

### DISCUSSION THREAD

**Discussion by FateRiddle** [xyflow/xyflow#1832] [Q&A] [answered]
I'm building a decision tree type of project, needs a ton of interactions. Tried this out, and finding `onElementClick` is the way to listen to click event on a node, and return that node's data to callback.

I may have to build a customNode, on which 2 or 3 buttons can be clicked, for adding, editing and deleting nodes, `onElementClick` will trigger on all clicks, and I can't distinguish which button is clicked based on the callback info.

Is there any way to solve or bypass this? Thanks!

**Comment by moklick** [comment, marked as answer]
When you are working with custom nodes you can define the click handlers inside the nodes. There is no need for the `onElementClick` handler anymore.

**Comment by imteekay** [comment]
Hi @moklick, can you expand this idea with an example? I'm searching the type definition for this click handle inside the node in the docs but didn't find it. Is there any prop for an element we can pass like a `onClick` or something like that?

Maybe like this:

```js
{
  id: '2',
  data: { label: <div onClick={() => alert('testing')}>Default Node</div> },
  position: { x: 100, y: 125 },
}
```
